### PR TITLE
nix-du: 0.3.3 -> 0.3.4

### PR DESCRIPTION
**Pull request by symphorien**
###### Motivation for this change

update (takes keep-output and keep-derivations into account)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119243` at 53502b3a run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>nix-du</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`lgpl3` is a deprecated license, please check if project uses `lgpl3Plus` or `lgpl3Only` and change `meta.license` accordingly.

Near pkgs/tools/package-management/nix-du/default.nix:29:5:
```
   |
29 |     license = licenses.lgpl3;
   |     ^
```
</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 119243` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nix-du</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package built (see https://github.com/NixOS/nixpkgs/pull/119243#issuecomment-818003332)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
..ew/pr-119243]$ ./results/nix-du/bin/nix-du
[...]
N1 -> N93;
N1 -> N94;
N1 -> N95;
N1 -> N96;
N1 -> N97;
}
```

Change looks good to me.

File: pkgs/tools/package-management/nix-du/default.nix [hunk 1]
```diff
@@ -1,17 +1,17 @@
-{ lib, stdenv, fetchFromGitHub, rustPlatform, nix, boost, graphviz, darwin }:
+{ lib, stdenv, fetchFromGitHub, rustPlatform, nix, boost, graphviz, darwin, pkg-config }:
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/nix-du/default.nix:1]:
```suggestion
{ lib, stdenv, fetchFromGitHub, rustPlatform, nix, boost, graphviz, Security, pkg-config }:
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/tools/package-management/nix-du/default.nix [hunk 2]
```diff
@@ -21,10 +21,12 @@ rustPlatform.buildRustPackage rec {
     nix
   ] ++ lib.optional stdenv.isDarwin darwin.apple_sdk.frameworks.Security;
```
[the remark this answers is not in the document]
Comment (symphorien) [reply pkgs/tools/package-management/nix-du/default.nix]: done

File: pkgs/tools/package-management/nix-du/default.nix [hunk 3]
```diff
@@ -1,30 +1,32 @@
...
+  cargoSha256 = "19fwkw9iswzkhqgfq7pmcabqmq5c7vvirwaxbfjshkwcgn47rgjl";
 
   doCheck = true;
   checkInputs = [ nix graphviz ];
 
   buildInputs = [
     boost
     nix
-  ] ++ lib.optional stdenv.isDarwin darwin.apple_sdk.frameworks.Security;
+  ] ++ lib.optional stdenv.isDarwin Security;
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/nix-du/default.nix:22]:
```suggestion
  ] ++ lib.optionals stdenv.isDarwin [ Security ];
```

**Pull request merged**